### DISCUSSION TITLE
Eventlet on Windows

### DISCUSSION THREAD

**Discussion by ramanraja** [eventlet/eventlet#759] [General]
Does Eventlet work on Windows platform? Are there any issues with threading vs forking ?

**Comment by temoto** [comment]
TL;DR: don't use eventlet on native windows. Do use eventlet in WSL, because it's basically linux.

General advice - if you can use modern python async IO, it's better choice than any light threading shim, like eventlet.

It works with couple of issues https://github.com/eventlet/eventlet/labels/platform-windows but only with `select` system call, which is the worst one of supported hubs.

In most cases eventlet does not use threads. `eventlet.tpool` is nice to extend existing app to use cores, but threads in python contend for GIL.

AFAIK, fork on windows is not available, and eventlet can't help with that.
